controllers/front/CartController.php

```diff
@@ -635,17 +646,26 @@ protected function areProductsAvailable()
         }
 
         if ($product['active']) {
-            return $this->trans(
-                '%product% is no longer available in this quantity. You cannot proceed with your order until the quantity is adjusted.',
-                ['%product%' => $product['name']],
-                'Shop.Notifications.Error'
+            if ((int) $product['quantity_available'] > 1 || (int) $product['quantity_available'] == 0) {
+                $sentence = 'There are only [1]%quantity%[/1] items of the product [1]%product%[/1] left in stock. Adjust the quantity in your cart to proceed with your order.';
+            } elseif ((int) $product['quantity_available'] == 1) {
+                $sentence = 'There\'s only [1]%quantity%[/1] item of the product [1]%product%[/1] left in stock. Adjust the quantity in your cart to proceed with your order.';
+            } else {
+                $sentence = 'The product [1]%product%[/1] in your cart is no longer available in this quantity. Adjust the quantity in your cart to proceed with your order.';
```

Comment:
@Hlavtox are you referring to this one?

Reply:
line 654

Reply:
Ah yes this must be the one

Reply:
@marsaldev, I see that you've added information about the remaining items. Don't you think it should depend on the setting available in BO? I didn't notice it before, but you've changed the behavior before it was unclear how many items are left in the stock. Now it's public, even if you don't want to.

Reply:
It was always there XD since July 22
I guess because at some point I was asked to me to have the singular and plural form